src/hazelweb/model/UndoHistory.rei

```diff
@@ -9,6 +9,8 @@ type undo_history_entry = {
   previous_action: Action.t,
   action_group: UndoHistoryCore.action_group,
   timestamp: UndoHistoryCore.timestamp,
+  current_cardstack: int,
```

Comment:
not sure I'm understanding how this connects to the `cardstacks_after_action` and `cardstacks_after_move`? actually I don't recall what the latter was for and it seems to be doing something buggy, e.g. type `abc`, then move to `ab|` and then do an undo and then a redo, and you end up at `ab|c` rather than `abc|`

Reply:
oh I think the intended behavior is that the redoing back to the top doesn't lose your cursor position. that isn't a thing that vs code does though, so we might want to drop that complexity.

in any case, doesn't this ZCardstacks.t contain the index of the current cardstack and card?